frontend/apps/web/app/(mgmt)/[account]/new/connection/postgres/PostgresForm.tsx

```diff
@@ -130,6 +136,66 @@ export default function PostgresForm() {
     }
   }
 
+  /* we call the underlying useGetConnection API directly since we can't call
+the hook in the useEffect conditionally. 
+*/
+  useEffect(() => {
+    const fetchData = async () => {
+      if (sourceConnId && account) {
+        setIsLoading(true);
+        try {
+          const response = await fetch(
+            `/api/accounts/${account?.id ?? ''}/connections/${sourceConnId}`,
+            {
+              method: 'GET',
+              headers: {
+                'Content-Type': 'application/json',
+              },
+            }
+          );
+          const data: GetConnectionResponse = await response.json();
+
+          const connData =
+            data instanceof GetConnectionResponse
+              ? data
+              : GetConnectionResponse.fromJson(data);
+
+          if (connData) {
+            const config = connData.connection?.connectionConfig?.config
+              .value as PostgresConnectionConfig;
+
+            const pgConfig = config.connectionConfig
+              .value as PostgresConnection;
+
+            setIsLoading(false);
+            // reset the form with the new values
+            form.reset({
+              ...form.getValues(),
+              connectionName: pgConfig.name + '_copy',
+              db: {
+                ...form.getValues().db,
+                host: pgConfig.host,
+                name: pgConfig.name,
+                user: pgConfig.user,
+                pass: pgConfig.pass,
+                port: pgConfig.port,
+                sslMode: pgConfig.sslMode,
+              },
+            });
+          }
+        } catch (error) {
+          console.error('Failed to fetch connection data:', error);
+        }
```

Comment:
You need to make sure that after all else, the isLoading state is set to false, otherwise the form is unusable. Add a `finally{}` block here to ensure that `setIsLoading` is always set to false. Then it does not need to be present in the try or catch blocks.